Code/Framework/AzToolsFramework/Tests/Prefab/PrefabUndoComponentPropertyOverrideTests.cpp

```diff
@@ -0,0 +1,287 @@
+
+        // Modify the local translation property in transform component as override
+        // Note: Entity needs to be modified manually since we would update cached instance DOM in undo node to avoid reloading
+        AZ::TransformBus::Event(wheelEntityInfo.m_entityId, &AZ::TransformInterface::SetLocalTranslation, overriddenTranslation);
+
+        PropertyChangePatch changePatch = MakeTransformTranslationPropertyChangePatch(wheelEntityInfo.m_entityId, overriddenTranslation);
+
+        PrefabDom propertyDomValue;
+        ConvertToPrefabDomValue(propertyDomValue, changePatch.m_propertyValue);
+
+        AZStd::string wheelEntityAliasPath = m_instanceToTemplateInterface->GenerateEntityAliasPath(wheelEntityInfo.m_entityId);
+        AZ::Dom::Path pathToPropertyFromOwningPrefab = AZ::Dom::Path(wheelEntityAliasPath) / changePatch.m_pathToProperty;
+
+        // Create an undo node
+        PrefabUndoComponentPropertyOverride undoNode("Modify transform translation");
+        InstanceOptionalReference carInstance = m_instanceEntityMapperInterface->FindOwningInstance(carInstanceInfo.m_containerEntityId);
+        EXPECT_TRUE(carInstance.has_value());
+
+        // Redo
+        undoNode.CaptureAndRedo(carInstance->get(), pathToPropertyFromOwningPrefab, propertyDomValue);
+        EXPECT_TRUE(undoNode.Changed());
+        PropagateAllTemplateChanges();
+
+        AZ::Vector3 currentWheelTranslate(-1.0f); // init to other value.
+        AZ::TransformBus::EventResult(currentWheelTranslate, wheelEntityInfo.m_entityId, &AZ::TransformBus::Events::GetLocalTranslation);
+        EXPECT_EQ(currentWheelTranslate, overriddenTranslation);
+        EXPECT_TRUE(
+            m_prefabOverridePublicInterface->AreOverridesPresent(wheelEntityInfo.m_entityId, changePatch.m_pathToProperty.ToString()));
+
+        // Undo
+        undoNode.Undo();
+        PropagateAllTemplateChanges();
+
+        currentWheelTranslate = AZ::Vector3(-1.0f); // reset
+        AZ::TransformBus::EventResult(currentWheelTranslate, wheelEntityInfo.m_entityId, &AZ::TransformBus::Events::GetLocalTranslation);
+        EXPECT_EQ(currentWheelTranslate, defaultTranslation);
+        EXPECT_FALSE(
+            m_prefabOverridePublicInterface->AreOverridesPresent(wheelEntityInfo.m_entityId, changePatch.m_pathToProperty.ToString()));
+
+        // Redo
+        undoNode.Redo();
+        PropagateAllTemplateChanges();
+
+        currentWheelTranslate = AZ::Vector3(-1.0f); // reset
+        AZ::TransformBus::EventResult(currentWheelTranslate, wheelEntityInfo.m_entityId, &AZ::TransformBus::Events::GetLocalTranslation);
+        EXPECT_EQ(currentWheelTranslate, overriddenTranslation);
+        EXPECT_TRUE(
+            m_prefabOverridePublicInterface->AreOverridesPresent(wheelEntityInfo.m_entityId, changePatch.m_pathToProperty.ToString()));
+    }
+
+    TEST_F(PrefabUndoComponentPropertyOverrideTests, EditTranslationOnEntityInNestedPrefabSucceeds)
```

Comment:
This can be added later but one more good test case can be creating the super car prefab hierarchy and focusing on the supercar and making it override the car prefab. It'll verify that overrides work not just from the topmost(level) prefab but will work from nested prefabs too.

Reply:
I will keep this test in this PR and leave this thread open. Can add it later.